docker/docker_image_src_test.go

```diff
@@ -22,3 +25,29 @@ func TestSimplifyContentType(t *testing.T) {
 		assert.Equal(t, c.expected, out, c.input)
 	}
 }
+
+type testClient struct{}
+
+func (c *testClient) MakeRequest(method, url string, headers map[string][]string, stream io.Reader, resolved bool) (*http.Response, error) {
+	return nil, nil
+}
+
+func TestIntendedDockerReference(t *testing.T) {
+	client := &testClient{}
+
+	img := "localhost:5000/runcom/busybox"
+	src, err := NewDockerImageSource(img, client)
+	assert.NoError(t, err)
+	expected := fmt.Sprintf("%s:latest", img)
+	assert.Equal(t, expected, src.IntendedDockerReference())
+
+	imgWithTag := "localhost:5000/runcom/busybox:amd64"
+	src, err = NewDockerImageSource(imgWithTag, client)
+	assert.NoError(t, err)
+	assert.Equal(t, imgWithTag, src.IntendedDockerReference())
+
+	dockerImg := "runcom/busybox:amd64"
+	src, err = NewDockerImageSource(dockerImg, client)
+	assert.NoError(t, err)
+	assert.Equal(t, dockerImg, src.IntendedDockerReference())
```

Comment:
Yeah, sure, it is clear that it can be used for some/most of the tests. But what it would really look like depends on how it is used in anger (e.g. testing the ping, http/https detection, authentication?).